### ISSUE TITLE
Upgrade to Kubernetes 1.24

### ISSUE THREAD

**Issue by lemeurherve** [jenkins-infra/helpdesk#3387]
Previous upgrade: https://github.com/jenkins-infra/helpdesk/issues/3053

- [x] Upgrade kubectl within [docker-helmfile](https://github.com/jenkins-infra/docker-helmfile)
   - [x]  First upgrade the updatecli manifest : https://github.com/jenkins-infra/docker-helmfile/pull/266
   - [x] Then apply the updatecli PR that is produced by this change: https://github.com/jenkins-infra/docker-helmfile/pull/267
   - [x] Then Apply the Kubernetes-Management PR that will use this new version - https://github.com/jenkins-infra/kubernetes-management/pull/3611

- [x] Send an email (ideally 1-2 day priori to the operations) on both jenkinsci-dev and jenkins-infra mailing lists (google groups) to let users known "when" the operations will take place, and "what" will be the expected impacts
  - https://groups.google.com/g/jenkinsci-dev/c/pmWN0Bhl5Sg 

- [x] Upgrade DOKS (`doks` and `doks-public`)
  - [-] Open datetime announcement on status.jenkins.io for `doks-public`
  - [-] Disable cluster in ci.jenkins.io configuration as code
  - [x] Disable kubernetes management (disable job on infra.ci.jenkins.io) 
  - [x] Bump Kubernetes version in jenkins-infra/digitalocean
  - [x] Ensure cluster is ~~re-created by terraform in jenkins-infra/digitalocean~~ updated in DigitalOcean management interface
  - [x] Re-Enable kubernetes management (and update kubeconfigs in the infra.ci's SOPS credentials (private repo), if kubernetes management cannot connect) and run a job on the main branch
  - [x] Add the credential of the service account `jenkins-agents` in ci.jenkins.io if needed (manual step) `doks` only not `doks-public`
  - [-] Close announcement

- [x] Upgrade EKS (`cik8s` and `eks-public`)
  - [x] Open datetime announcement on status.jenkins.io - https://github.com/jenkins-infra/status/pull/245
  - [x] Disable cluster in ci.jenkins.io configuration as code - https://github.com/jenkins-infra/jenkins-infra/pull/2653
  - [x] Disable kubernetes management (disable job on infra.ci.jenkins.io)
  - [x] Upgrade cluster through Terraform (control plane + node groups) - https://github.com/jenkins-infra/aws/pull/342
	  - [x] add-ons manually updated (see https://github.com/jenkins-infra/aws/issues/278 for next k8s upgrades)
  - [x] Ensure the upgraded cluster is still OK (check kubernetes-management and run a job on the main branch)
	⚠️ Ensure that the security group `eks-cluster-sg-<cluster name>-<random ID>` that is managed directly by EKS doesn't contain the tag `kubernetes.io/cluster/<cluster name>`
  	Even with `create_cluster_primary_security_group_tags = false` this tag is wrongly set by the terraform module and must be manually deleted afterwards for now.
  
	![image](https://user-images.githubusercontent.com/91831478/220706518-ca617eca-eeaa-4e28-9678-3d31c5fa1df0.png)

  - [x] Enable it again in ci.jenkins.io config as code
  - [x] Close announcement
 
- [x] Upgrade AKS (`privatek8s`, `publick8s` and `prodpublick8s`)
  - [-] ~Check [previous AKS upgrades](https://github.com/jenkins-infra/documentation/blob/main/maintenance/kubernetes/aks-1.23.md) and create one for 1.24~
  - [x] Open datetime announcement on status.jenkins.io
  - [x] Disable kubernetes management (disable job on infra.ci.jenkins.io)
  - [x] Upgrade privatek8s first
  	⚠️ It will put infra.ci out of service so might need manual steps to re-boostrap the kubernetes-management job
	  - [x] Version retrieved with `az aks get-upgrades --resource-group prod-privatek8s --name privatek8s-emerging-ram
--output table`
	  - [x] Control plane upgraded with terraform
	  - [x] Nodes upgraded via the Azure Portal (for now)
	  		
		<details>
		<summary>Details</summary>
		
		<img width="1433" alt="image" src="https://user-images.githubusercontent.com/91831478/220934415-20d0e2ec-eb3b-48a3-8202-9de7405cc563.png">
		
		</details>

  - [x] Upgrade publick8s
  - [x] Upgrade prodpublick8s
  - [x] Re-Enable kubernetes management (and update kubeconfigs in the infra.ci's SOPS credentials (private repo), if kubernetes management cannot connect) and run a job on the main branch
  - [x] Send an email on both jenkinsci-dev and jenkins-infra mailing lists to announce the end of maintenance
  - [x] Close announcement

**Comment by dduportal** [comment]
We had an issue while upgrading `cik8s` on EKS with the following error message during the addons upgrade:

```
error updating EKS Add-On (<redacted>:aws-ebs-csi-driver): AccessDeniedException: User is not authorized to perform this action
```

Thanks to @lemeurherve 's researches, we found https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1725 which confirms that the Terraform EKS module is (was?) not exhaustive for the IAM `eks` required permissions.
We covered https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonelastickubernetesservice.html#amazonelastickubernetesservice-actions-as-permissions exhaustively to apply the missing permissions (mainly: `eks:UpdateAddon` 😅 but not only).

Applied the new permissions in the (private) repository jenkins-infra/terraform-state (ref. TBD) and triggered a new (successful) build.
